dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+     */
+    @Override
+    public Mono<BulkResponse> bulk(HttpHeaders headers, BulkRequest bulkRequest) {
+        return sendRequest(bulkRequest, DotReactiveElasticsearchClient.RequestCreator.bulk(), BulkResponse.class, headers) //
+                .publishNext();
+    }
+
+    // --> INDICES
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#existsIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.get.GetIndexRequest)
+     */
+    @Override
+    public Mono<Boolean> existsIndex(HttpHeaders headers, GetIndexRequest request) {
+
+        return sendRequest(request, DotReactiveElasticsearchClient.RequestCreator.indexExists(), RawActionResponse.class, headers) //
+                .map(response -> response.statusCode().is2xxSuccessful()) //
+                .next();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#deleteIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest)
+     */
+    @Override
+    public Mono<Void> deleteIndex(HttpHeaders headers, DeleteIndexRequest request) {
+
+        return sendRequest(request, DotReactiveElasticsearchClient.RequestCreator.indexDelete(), AcknowledgedResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#createIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.create.CreateIndexRequest)
+     */
+    @Override
+    public Mono<Void> createIndex(HttpHeaders headers, CreateIndexRequest createIndexRequest) {
+
+        return sendRequest(createIndexRequest, DotReactiveElasticsearchClient.RequestCreator.indexCreate(), AcknowledgedResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#openIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.open.OpenIndexRequest)
+     */
+    @Override
+    public Mono<Void> openIndex(HttpHeaders headers, OpenIndexRequest request) {
+
+        return sendRequest(request, DotReactiveElasticsearchClient.RequestCreator.indexOpen(), AcknowledgedResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#closeIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.close.CloseIndexRequest)
+     */
+    @Override
+    public Mono<Void> closeIndex(HttpHeaders headers, CloseIndexRequest closeIndexRequest) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'headers' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)